content/posts/infrastructure-as-code/index.md

```diff
@@ -0,0 +1,438 @@
+---
+title: Infrastructure as Code
+author: Lena Fuhrimann / Mathis Kretz
+comments: true
+date: 2021-12-09
+---
+
+This blog post is the second part of a three-part series adapted from a GitOps webinar series which we co-produced together with our friends at [VSHN](https://www.vshn.ch/).
+
+In this second part we will look into a topic that is crucial to any modern cloud-native environment in order to provide stability and parity across environments: Infrastructure as Code. We will delve into some interesting tools in that realm and compare different concepts and configurations around how to deploy and maintain infrastructure in a sane way. After reading the blog post, you will have a good understanding of IaC and its benefits and you’ll have a mature toolset available in order to implement it in your team.
+
+If you have questions, feel free to post them as comments on this blog post. If you would rather sit back and enjoy this part as a webinar, then you may head over to [the recording on Youtube](https://youtu.be/OrGVV8q8yeo).
+
+# Introduction
+
+Before we dive into some technical solutions and frameworks for Infrastructure as Code, let’s first examine what we mean by that term.
+
+## What is not Infrastructure as Code?
+
+Or actually, let’s start off by looking at what we would not call Infrastructure as Code. While certainly being awesome in their own right, we would not consider as Infrastructure as Code tools like Ansible, Puppet and Chef which allow you to apply configurations to existing VMs. The reason for this is that these tools generally do not take care of the VM or resource creation step itself which, as you will see later, we consider an important part of Infrastructure as Code. That said, tools for VM configuration certainly have their place in DevOps as we will also see when we have a closer look at Packer.
+
+Furthermore, we would also not consider imperative scripts which create VMs (and possibly install things on them) as Infrastructure as Code. This is because we consider declarativeness as an important feature of Infrastructure as Code as you will also see later.
+
+## What is Infrastructure as Code?
+
+```terraform
```

Comment:
I would put `hcl`. Even though, `terraform` is aliased and works, `hcl` is more correct because that's the name of the language.

Reply:
Good point, thanks! Done!